src/Grpc.Core.Api/Utils/EncodingExtensions.cs

```diff
@@ -0,0 +1,54 @@
+#region Copyright notice and license
+// Copyright 2019 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Text;
+
+namespace Grpc.Core.Api.Utils
+{
+
+    internal static class EncodingExtensions
```

Comment:
What uses this?

Reply:
added a TODO to revisit. This is probably used by Grpc.Core, so we'd have to make changes there first to avoid breaking it.